craft_application/services/remotebuild.py

```diff
@@ -249,15 +251,18 @@ def _ensure_project(self) -> launchpad.models.Project:
                 summary=f"Automatically-generated project for housing {self.lp.username}'s remote builds in snapcraft, charmcraft, etc.",
             )
 
-    def _new_repository(
+    def _ensure_repository(
         self, project_dir: pathlib.Path
     ) -> tuple[WorkTree, launchpad.models.GitRepository]:
-        """Create a repository on the local machine and on Launchpad."""
+        """Create a repository on the local machine and ensure it's on Launchpad."""
         work_tree = WorkTree(self._app.name, self._name, project_dir)
         work_tree.init_repo()
-        lp_repository = launchpad.models.GitRepository.new(
-            self.lp, self._name, target=self._lp_project.name
-        )
+        try:
+            lp_repository = self.lp.new_repository(self._name, self._lp_project.name)
+        except launchpadlib.errors.HTTPError:
+            lp_repository = self.lp.get_repository(
+                name=self._name, project=self._lp_project.name
+            )
```

Comment:
I think we need to check what kind of error here?

Reply:
I don't think so. Any error other than "the repository already exists" will also result in an appropriate error when trying to retrieve the repository